app/tooloptionwidget.cpp

```diff
@@ -47,14 +47,34 @@ void ToolOptionWidget::updateUI()
     mPreserveAlphaBox->setVisible( currentTool->isPropertyEnabled( PRESERVEALPHA ) );
     mUseAABox->setVisible(currentTool->isPropertyEnabled( ANTI_ALIASING ) );
     mInpolLevelsBox->setVisible(currentTool->isPropertyEnabled( INTERPOLATION ) );
+    mToleranceSlider->setVisible(currentTool->isPropertyEnabled( TOLERANCE ) );
+    mToleranceSpinBox->setVisible(currentTool->isPropertyEnabled( TOLERANCE ) );
+    mFillContour->setVisible( currentTool->isPropertyEnabled( FILLCONTOUR ) );
 
     auto currentLayerType = editor()->layers()->currentLayer()->type();
 
     if(currentLayerType == Layer::VECTOR)
     {
-        mVectorMergeBox->setVisible( currentTool->isPropertyEnabled( VECTORMERGE) );
+        // TODO: implement merge behavior
+//        mVectorMergeBox->setVisible( currentTool->isPropertyEnabled( VECTORMERGE) );
+        mToleranceSlider->setVisible(false);
+        mToleranceSpinBox->setVisible(false);
+        mUseAABox->setVisible(false);
+
+        if (currentTool->type() == PENCIL)
+        {
+            mSizeSlider->setVisible(false);
+            mBrushSpinBox->setVisible(false);
+            mUsePressureBox->setVisible(false);
+        }
+    } else {
+        if (currentTool->type() == PENCIL)
+        {
+            mFillContour->setVisible(false);
+        }
```

Comment:
Probably we need to find a way to organise all these visible/invisible switchings.